### PR TITLE
EPMRPP-49152: [AGENT] Jasmine. Reporting Before and After hooks

### PR DESCRIPTION
**Pull request by KatsiarynaTatarynovich**

### CODE REVIEW

File: examples/protractor/testGithubPage.js [hunk 1]
```diff
@@ -30,6 +30,23 @@ const attachments = [
...
+    let originalTimeout;
+
+    beforeAll(function() {
+        const func = () => {};
+
+        func();
+    });
+
+    beforeEach(function() {
+        originalTimeout = jasmine.DEFAULT_TIMEOUT_INTERVAL;
```
Comment (AmsterGet) [inline examples/protractor/testGithubPage.js:42]: Which purpose of it?

File: lib/jasmine-reportportal-reporter.js [hunk 2]
```diff
@@ -162,6 +178,38 @@ class ReportportalReporter {
         promiseErrorHandler(stepObj.promise);
     }
 
+    hookStarted(hook) {
+        const parent = this.getParentId(2);
```
Comment (AmsterGet) [inline lib/jasmine-reportportal-reporter.js:182]: Magic number.
What does it mean?

File: lib/jasmine-reportportal-reporter.js [hunk 3]
```diff
@@ -197,13 +245,62 @@ class ReportportalReporter {
...
 
     suiteDone() {
         let suiteDonePromise = this.client.finishTestItem(this.getParentId(), {});
 
         promiseErrorHandler(suiteDonePromise.promise);
         this.finishParent();
+        this.startTime = null;
+    }
+
+    makeWrapperAll(wrapped, funcStart, funcFinish) {
```
Comment (AmsterGet) [inline lib/jasmine-reportportal-reporter.js:259]: May this function be moved to another file?

File: lib/constants/itemTypes.js [hunk 4]
```diff
@@ -2,6 +2,22 @@ const entityType = {
...
+  AFTER_METHOD: 'AFTER_METHOD',
+};
+const hookTypes = {
+  BEFORE_ALL: 'beforeAll',
+  BEFORE_EACH: 'beforeEach',
+  AFTER_ALL: 'afterAll',
+  AFTER_EACH: 'afterEach',
+};
+const hookTypesMap = {
+  [hookTypes.BEFORE_EACH]: entityType.BEFORE_METHOD,
```
[the remark this answers is not in the document]
Comment (ElenaRomanchuk) [reply lib/constants/itemTypes.js:17]: I don't think so. Actually we don't have test at all, just suites and steps. So, I think beforeEach matches to beforMethod type.

File: examples/protractor/testGithubPage.js (hunk 1 above)
Comment (KatsiarynaTatarynovich) [reply examples/protractor/testGithubPage.js:42]: I have the Promise inside of the test and it causes some problems, this string solves the problems

File: lib/jasmine-reportportal-reporter.js [hunk 5]
```diff
@@ -17,21 +17,35 @@ class ReportportalReporter {
...
+    reportHooks() {
+        this.conf.reportHooks && this.installHooks();
     }
 
     escapeMarkdown(string) {
         return string.replace(/_/gm, '\\_').replace(/\*/gm, '\\*');
     }
 
-    getParentId() {
+    getParentId(number) {
```
Comment (ElenaRomanchuk) [inline lib/jasmine-reportportal-reporter.js:37]: Please, give meaningful name to number. Or maybe extract logic getting parent of parent into separate method.

File: lib/jasmine-reportportal-reporter.js (hunk 3 above)
Comment (KatsiarynaTatarynovich) [reply lib/jasmine-reportportal-reporter.js]: Yes, I moved it to utils

**Pull request merged**